pallets/cash/src/pipeline.rs

```diff
@@ -369,28 +369,30 @@ fn prepare_augment_cash<T: Config>(
     mut st: State,
     recipient: ChainAccount,
     principal: CashPrincipalAmount,
+    from_external: bool,
 ) -> Result<State, Reason> {
     let recipient_cash_pre = st.get_cash_principal::<T>(recipient);
 
     let (recipient_repay_principal, _recipient_supply_principal) =
         repay_and_supply_principal(recipient_cash_pre, principal)?;
 
     let recipient_cash_post = recipient_cash_pre.add_amount(principal)?;
-
-    let chain_id = recipient.chain_id();
-    let chain_cash_principal_post = st
-        .get_chain_cash_principal::<T>(chain_id)
-        .sub(principal)
-        .map_err(|_| Reason::NegativeChainCash)?;
-
     let total_cash_post = st
         .get_total_cash_principal::<T>()
         .sub(recipient_repay_principal)
         .map_err(|_| Reason::InsufficientChainCash)?;
 
     st.set_cash_principal::<T>(recipient, recipient_cash_post);
     st.set_total_cash_principal::<T>(total_cash_post);
-    st.set_chain_cash_principal::<T>(chain_id, chain_cash_principal_post);
+
+    if from_external {
```

Comment:
Are we allowed to bail after we set things here? Is that behavior used elsewhere?

Reply:
We don't write until commit right?

Reply:
Or not sure what you mean